### PR TITLE
POST - Building a REST API - Spring Boot vs Micronaut Framework

### PR DESCRIPTION
**Pull request by sdelamo**
`./gradlew buildingARestApiSpringBootVsMicronautPostBuild`

### CODE REVIEW

File: guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+|===
+
+=== Spring Boot Controller
+
+source:SaasSubscriptionPostController[app=springboot]
+
+callout:spring-boot-rest-controller[1]
+callout:spring-boot-request-mapping[2]
+callout:constructor-di[number=3,arg0=SaasSubscriptionRepository]
+callout:spring-boot-post-mapping[number=4,arg0=findById,arg1=/subscriptions/{id}]
```
Comment (timyates) [inline guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc]: This renders

> The @PostMapping annotation maps the findById method to an HTTP POST request on /subscriptions/{id}.

Which is not what this method is doing...

Should it be:

> The @PostMapping annotation maps this method to a POST request.

?

File: guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+callout:spring-boot-request-body[5]
+
+
+=== Micronaut Controller
+
+source:SaasSubscriptionPostController[app=micronautframework]
+
+callout:controller[number=1,arg0=/subscriptions]
+callout:constructor-di[number=2,arg0=SaasSubscriptionRepository]
+callout:post[number=3,arg0=findById,arg1=/subscriptions/{id}]
```
Comment (timyates) [inline guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc]: This renders

> The [@Post](https://docs.micronaut.io/latest/api/io/micronaut/http/annotation/Post.html) annotation maps the findById method to an HTTP POST request on /subscriptions/{id}.

Again, this is not what this method is doing...

Should it be:

> The [@Post](https://docs.micronaut.io/latest/api/io/micronaut/http/annotation/Post.html) annotation maps this method to a POST request.

?

**Review by timyates — CHANGES_REQUESTED** [review]

File: guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc (hunk 1 above)
Comment (sdelamo) [reply guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc]: resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1457/commits/eece9ef15db2941179d5aeb6fae45aed3701ab58

File: guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc (hunk 2 above)
Comment (sdelamo) [reply guides/building-a-rest-api-spring-boot-vs-micronaut-post/building-a-rest-api-spring-boot-vs-micronaut-post.adoc]: resolved via resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1457/commits/eece9ef15db2941179d5aeb6fae45aed3701ab58

**Review by timyates — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

**Pull request merged**